src/lib/battery/battery.h

```diff
@@ -201,4 +203,12 @@ class Battery : public ModuleParams
 	float _scale{1.f};
 	uint8_t _warning{battery_status_s::BATTERY_WARNING_NONE};
 	hrt_abstime _last_timestamp{0};
+
+	float first_run_time;
```

Comment:
Note - `first_run_time` should be of type `hrt_abstime` (which is `uint64_t`) and initialised to zero.
Risk of mixing types (am guessing any `hrt_abstime - float` will default to `float`, which may lack precision).  Probably not a big issue as I think PX4 time is micro-seconds, just good practice not to mix types unless there's a good reason.